ee/api/billing.py

```diff
@@ -181,6 +183,22 @@ def list(self, request: HttpRequest, *args: Any, **kwargs: Any) -> Response:
         if org:
             self._update_org_details(org, response)
 
+        # check if the distinct_ids are in sync, if not patch the billing service
+        received_distinct_ids = response.get("distinct_ids") or []
+        local_distinct_ids = []
+        if is_cloud() and org:
+            local_distinct_ids = list(org.members.values_list("distinct_id", flat=True))
+        else:
+            local_distinct_ids = list(User.objects.values_list("distinct_id", flat=True))
+
+        if set(received_distinct_ids) != set(local_distinct_ids):
+            billing_service_token = build_billing_token(license, org)
+            requests.patch(
+                f"{BILLING_SERVICE_URL}/api/billing/",
+                headers={"Authorization": f"Bearer {billing_service_token}"},
+                json={"distinct_ids": local_distinct_ids},
+            )
```

Comment:
Don't love this... Getting the billing info triggers a patch request.

Could we not include the distinct_ids in the token instead? That way the billing service can just check the token and update that way?

Reply:
I like that, smart friend, let me do it